### DISCUSSION TITLE
$2

### DISCUSSION THREAD

**Discussion by gerontakos** [crystalyragui/MARC2RDA#359] [General]
$2 previously discussed at:
[$0 and $1 discussion #328](https://github.com/uwlib-cams/MARC2RDA/discussions/328#discussioncomment-2374443)
[$0 and $1 discussion #328](https://github.com/uwlib-cams/MARC2RDA/discussions/328#discussioncomment-2425973)
[$0 and $1 discussion #328](https://github.com/uwlib-cams/MARC2RDA/discussions/328#discussioncomment-2465466)

**Comment by tmqdeborah** [comment]
### 33X $2 Source codes with appended language codes

We have found some examples in the UW file where $2 (source) is, e.g.,

336 $6 880-09 $a Wen zi $b txt $2 rdacontent/chi
337 $6 880-10 $a Wu mei jie $b n $2 rdamedia/chi
338 $6 880-11 $a Cheng ce $b nc $2 rdacarrier/chi


  336 $a text $b txt $2 rdacontent/chi
  337 $a unmediated $b n $2 rdamedia/chi
  338 $a volume $b nc $2 rdacarrier/chi


  336 $a text $b txt $2 rdacontent/cze
  337 $a bez média $b n $2 rdamedia/cze
  338 $a svazek $b nc $2 rdacarrier/cze

  336 $a Text $b txt $2 rdacontent/ger
  337 $a Computermedien $b c $2 rdamedia/ger
  338 $a Online-Ressource $b cr $2 rdacarrier/ger

 The RDA value vocabularies at https://www.rdaregistry.info/termList/ provide translations in multiple languages. It appears that if an extension to an id.loc source code is a MARC language code, that is meant to indicate that the term provided in a $a is in the language indicated by the MARC language code. However, as far as we can tell, id.loc does not provide a list of those terms in those languages.
 
 The translations of the RDA value vocabularies are available as downloads, and our project has used them for our lookup tables for our mapping, i.e., terms are available in all translated languages.
 
The best way to handle these extensions seems to be:
  - If a field carries RDA vocabulary values, e.g., 33X, 34X, and any associated 800 fields
    - If a $2 is present and is an rda or lcrda source code, e.g., 'rdaco' (RDA) or 'rdacontent' (LC) 
      - and no extension is present, e.g., '/chi', 
        - AND $a contains a term in any language that matches a term in the RDA lookup file, 
          - THEN map the term as an appropriate RDA IRI, e.g., http://rdaregistry.info/termList/RDAContentType/1020
    - If a $2 is present and is an rda or lcrda source code, e.g., 'rdaco' (RDA) or 'rdacontent' (LC) 
      - and an extension is present, e.g., '/chi', 
        - AND $a contains a term in any language that matches a term in the RDA lookup file, 
          - THEN map the term as an appropriate RDA IRI, e.g., http://rdaregistry.info/termList/RDAContentType/1020
    - If a $2 is present and is an rda or lcrda source code, e.g., 'rdaco' (RDA) or 'rdacontent' (LC) 
      - and an extension is present, e.g., '/chi', 
        - AND $a contains a term in any language that does not match a term in the RDA lookup file, 
          - THEN mint each $a value as a Concept -- we cannot tell whether a value of $a has the same meaning as RDA if we do not have a $2 source that we can use

Is this an acceptable solution?

**Reply by GordonDunsire** [reply]
@tmqdeborah: I think the extensions can be ignored. I don't see what benefit is gained from the last option, to mint a concept if the value of subfield $a is not found in the stated RDA vocabulary (including all language versions).  Instead, I think the string value should be what is output. The only MARC 21 data that may be lost is what is implied by the possibly undocumented use of language code extensions. Otherwise, we have agreed that the transform should not correct 'incorrect' values.

Your decision tree can be simplified to:

For MARC 21 fields that accommodate an RDA value vocabulary:

If a subfield $2 is present and has the value of an rda or lcrda source code, with or without a language extension:
 Look up the value of subfield $a in the RDA lookup file. If a match is found:
   Output the value of the RDA IRI
 Else
  Output the value of subfield $a.

The output does not require data provenance because either a) the IRI does not need it or b) the string has already been checked and its provenance cannot be an RDA value vocabulary (I am ignoring the additional complexity of version control).

**Reply by tmqdeborah** [reply]
@GordonDunsire, thanks for the feedback.

Minting a Concept for a vocabulary term with a $2 but without an approved source seems to be an established thing. See, e.g., [336 spreadsheet](https://docs.google.com/spreadsheets/d/1DuGquVXTFgm-_S9R4LHv9Olcf_GAa_EathimuxHBlcc/edit?pli=1&gid=1939665718#gid=1939665718&range=34:34), [337 spreadsheet ](https://docs.google.com/spreadsheets/d/1lI7Kly3Qw-rkdm_xFpZOipVpTSvZSt7ggpLiEfu5EzM/edit?gid=209962246#gid=209962246&range=5:5), [338 spreadsheet](https://docs.google.com/spreadsheets/d/1AjydEiUWn-fZfjk5nhgWKwJHWnwlcnMHgdZnLbYyZtU/edit?gid=1719915577#gid=1719915577&range=5:5)

And I found this quote in Issue #[439 ](https://github.com/uwlib-cams/MARC2RDA/issues/439#issuecomment-2476991289)
> [cspayne](https://github.com/cspayne)
[on Nov 14, 2024](https://github.com/uwlib-cams/MARC2RDA/issues/439#issuecomment-2476991289)
Our decision here should be added to the Decisions index. For vocabulary terms such as in 3XX fields, we perform lookups to retrieve the IRI for that term if possible based on the source. Otherwise, we mint a skos:Concept if a source is provided that we cannot do a lookup for. If there is no source, we use a string value. 

I wasn't present at the meeting where this was decided, but maybe it was decided this way in order to make it easier for doing cleanup by providing the unapproved or erroneous source. It does make it easier to see what is going on when we can see that a term that is not valid for, e.g., Carrier type, has "inScheme: rdacontent" and is actually a term that is valid for Content type (it could have been in a 338 (carrier) field instead of the 336 (content) field). 

If this isn't really necessary/helpful **for vocabulary terms** then it would certainly cut down on the number of triples that we are creating. But making this change would, unfortunately, mean changing a considerable amount of coding because there are a lot of 'vocabulary' fields with "Source of term" data.

So, I'm inclined to say that we should (for now) continue with the decision to:  "perform lookups to retrieve the IRI for that term if possible based on the source. Otherwise, we mint a skos:Concept if a source is provided that we cannot do a lookup for. If there is no source, we use a string value".

And maybe someone could (sometime) go through the Issues, Discussions, Meeting notes, and Decision index and find everything related to 'source' data (in $2 or Indicators) for Identifiers, classifications numbers, subject headings and terms and compile the decisions for each type of data in one document, and maybe get AI to provide a summary of our decisions :-) This could be very useful for checking consistency in the coding and for sharing with others in our final report.

And then, perhaps we could revisit this decision in Phase II. 

What do others think?

**Reply by GordonDunsire** [reply]
@tmqdeborah: 'if possible' means if the source has a means of look-up; it does not mean 'lookup failure'. The reason for minting a concept for 'if not possible' sources is so we can make a statement about the source via the concept IRI. If there is a 'lookup failure' we do not want to make a statement about the source because the term is not in the source.

**Reply by tmqdeborah** [reply]
@GordonDunsire, That sounds logical. 

However, it is very easy for a cataloger to get mixed up and give the wrong rda or lcrda source code for a term or code that is valid for a field and even give the wrong field for a valid term or code, especially between the 336 fields. 

Given that, what would you think of us doing this:
  - If a subfield $2 is present and has the value of an rda or lcrda source code for either 336, 337, or 338, with or without a language extension, and the value term or code in either a $a or $b is correct for any of those fields:
    - Look up the value in the RDA lookup files for all 3 fields. 
      - If a match is found in any of those lookup files for any value in a $a or $b
        - Output the RDA IRI for the value
      - Else
        - Output the value of subfield $a as a string value

Here are some real examples (hopefully with no errors on my part re how they will map):

- 338    $a computer $2 rdamedia -- valid term and source but wrong field: will map to RDA URI for term (media type)

- 338    $a microform $2 local -- valid term and field but $2 not rda: will map $a as concept, for field value (carrier type)

- 336    $a text $b txt -- valid term and code and field but no $2: will map $a and $b as string values for field value (content type)

- 338    $a computer $b c -- valid term and code but wrong field and no $2: will map $a and $b as string values for field value (carrier type)

- 338    $a text $2 rdacarrier -- valid term, wrong field and wrong source: will map to RDA URI for term value (content type)

- 336    $a volume $b nc $2 rdacarrier -- valid term, code, and source but wrong field: will map to RDA URI for term and code (carrier type)

- 336    $a video $b v $2 rdamedia -- valid term, code, and source but wrong  field: will map to RDA URI for term and code (media type)

- 337    $a cartographic image $b cri $2 rdacontent -- valid term, code, and source but wrong field: will map to RDA URI for term and code  (content type)

- 337    $a online resource $b cr $2 rdacarrier -- valid term, code, and source but wrong field: will map to RDA URI for term and code (carrier type)

Can you see any problem with doing this?

I don't think this kind of mix-up would happen as often with the 34X - 38X fields.

If you don't think this is a good idea, then we should add this to our list of suggested preprocessing fixes prior to running the transform.

**Reply by GordonDunsire** [reply]
@tmqdeborah: you are extending the scope of this sub-topic to subfield $b and to mis-assignment of valid and invalid values to the MARC 21 fields for RDA/ONIX vocabularies?

The original scope (subfield $a and subfield $2 lookup) applies to all MARC 21 fields that carry controlled vocabulary terms or codes. It is not confined to the RDA vocabularies. I think the project agreed in the past not to use a lookup function in the transform because it would slow it down too much. For example, we discussed the feasibility of lookup of a string value when we know the source provides IRIs (and @Adam included this in his analysis for name heading sources).

If we are treating the RDA vocabularies as exceptions (for obvious reasons), then:

- it is desirable to execute one lookup function per value of the term or code for each occurrence of MARC 21 336, 337, and 338 fields.
- to extend this to trapping field mis-assignment errors requires three lookup functions per value of the term or code.
- the number of lookups can be reduced by combining the RDA vocabularies into a single lookup vocabulary.
- a combined RDA/ONIX lookup vocabulary can only be used to lookup a term; the notations/codes are only unique to each RDA vocabulary, but the terms are 'globally' unique within the English version of RDA (the Glossary shows this).

I don't know if the terms are globally unique within other language versions of RDA, although they should be.
However, it is risky to use this approach across multiple language versions of RDA because there is constraint on the same string being used to mean different things in different languages. We are back to the original topic :-)

If the lookup file includes the terms and codes from LoC sources, the codes are distinct from RDA notations (alpha vs numeric) and should be 'globally' unique between the LoC sources. I don't know if all of the LoC terms are identical to the RDA/ONIX terms.

A combined English RDA/LoC lookup file could include the RDA maps from LoC to RDA. This would be a neat output of the project. Or does RV already have it?

**Comment by AdamSchiff** [comment]
If we could get the OCLC #s or the MMS IDs I can look at them and see what's going on.

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
aschiff @ uw.edu
________________________________

**Reply by tmqdeborah** [reply]
Here are the ones I found in the small file (150K) that I have for UW. 
[33X$a_slash.txt](https://github.com/user-attachments/files/21024176/33X.a_slash.txt)

**Comment by AdamSchiff** [comment]
Thanks!

OCLC #1389729985 is a record from a Czech library, used in Alma as an order record.  The book is not cataloged by us yet and this record would be replaced by one with 040 $b eng

Also, interestingly, there is no Czech in the RDA registry for these terms, so neither $2 rdaco/cze nor rdacontent/cze seems to be actually correct. However the $b codes are not language dependent, and since they come from the LC list I guess rdacontent/cze etc. is partially correct.  Not sure whether $2 should or should not be deleted, but I've left them alone for now.

The inclusion of this record in our set to transform seems wrong. It indicates to me that we should not be including order records for things that we haven't yet received or that haven't been cataloged and given a call number and item record yet from the records that we transform.

OCLC #1377673351 is also not a record that we have our holdings on in OCLC and is another order record where 040 $b chi makes this record not something we will ultimately be using once the book is cataloged.  This is like the one above in that the codes in $b do come from the LC vocabularies, so I've left them alone. I don't know why transliterated fields are given at all though in a record with 040 $b chi.

In OCLC #1227052842 the 040 $b is chi but the terms in 336-338 are in English so /chi shouldn't have been included in the $2. I've deleted the language qualifier.  This is also an order record that we would not be using when the cataloging is completed.

#1191859076 is the same as the one above. Except for the fact that I don't see this OCLC # in Alma at all.  But when I search by the MMS ID, it too is an order record.  I removed /chi from the 336-338.

#1220976259, #1264129539 and #1256977536 are same as above as well. 040 $b chi and an order record.  I removed /chi from the 336-338.

#695716404 has 040 $b ger but it isn't strictly German language of cataloging anymore. OCLC numerous times has futzed with it. Most of the notes are in English:

040  CHVBK ǂb ger ǂe pn ǂc CHVBK ǂd OCLCQ ǂd OCLCF ǂd OCLCO ǂd OCL ǂd OCLCQ ǂd OCLCO ǂd OCLCL ǂd OCLCQ
043  e-ne---
090  ǂb
049  WAUW
24504The confession or declaration of the ministers or pastors, which in the United Provinces are called Remonstrants : ǂb concerning the chief points of Christian religion.
260  London : ǂb Printed for Francis Smith at the Elephant and Castle in Cornhil near the Royal Exchange, ǂc 1676.
300  1 online resource (Online-Datei ([12], 264 Seiten))
336  Text ǂb txt ǂ2 rdacontent/ger
337  Computermedien ǂb c ǂ2 rdamedia/ger
500  Errata: prelim. p. [8].
504  Includes bibliographical references.
500  A translation of: Confessio sive declaratio sententiae pastorum qui in Foederato Belgio Remonstrantes vocantur super praecipuis articulis religionis Christianae.
500  Reproduction of original in the Folger Shakespeare Library, Washington, D.C.
500  Originaltitel: Confessio sive declaratio sententiae pastorum qui in Foederato Belgio Remonstrantes vocantur super praecipuis articulis religionis Christianae.
5880 Print version record.

Since this has now mostly English notes, I have changed the 336-337 to English, and deleted the 500 note in German since it's covered by another 500 note.  I changed the pagination to English too.  The record also didn't match the NAR set up for this expression and I added a 110 and 240 to it.


I will work my way through the remainder as time permits, but it's clear that all of these were incorrect in some way.  And we shouldn't transform uncataloged resource that we are just using records with 040 $b not = eng as temporary order records.

Adam


Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
aschiff @ uw.edu

________________________________
